homeassistant/components/zeroconf/__init__.py

```diff
@@ -37,11 +43,55 @@
 
     await zeroconf.register_service(info)
 
-    async def stop_zeroconf(event):
+    async def new_service(service_type, name):
+        """Signal new service discovered."""
+        service_info = await zeroconf.get_service_info(service_type, name)
+        info = info_from_service(service_info)
+        print(name, info)
+
+        for domain in manifest.SERVICE_TYPES[service_type]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    domain, context={'source': DOMAIN}, data=info
+                )
+            )
+
+    def service_update(_, service_type, name, state_change):
+        """Service state changed."""
+        from aiozeroconf import ServiceStateChange
```

Comment:
This can be nowadays on top of the file, right?

Reply:
Isn't that only for config entries?

Reply:
Can be at the top 👍  We install requirements based on manifest.json before importing any Python file.